### PR TITLE
Helm 2,3 supports CRD at the same time

### PR DESCRIPTION
**Pull request by tomahawk28**
Hi, 
This is the same PR we open for supporting helm 2 & 3 at the same time what we have. 
https://github.com/giantswarm/velero-app/pull/3

After patching this, the helm would use CRD in one directory.

### CODE REVIEW

**Comment by tomahawk28** [comment]
Would you mind to take a look? cc> @carlisia, @cpanato

File: charts/velero/templates/backupstoragelocation.yaml [hunk 1]
```diff
@@ -3,6 +3,9 @@ apiVersion: velero.io/v1
 kind: BackupStorageLocation
 metadata:
   name: {{ include "velero.backupStorageLocation.name" . }}
+  annotations:
+    "helm.sh/hook": post-install
```
Comment (jenting) [inline charts/velero/templates/backupstoragelocation.yaml:7]: A question, why this not pre-install but post-install instead? [1]

File: charts/velero/templates/volumesnapshotlocation.yaml [hunk 2]
```diff
@@ -3,6 +3,9 @@ apiVersion: velero.io/v1
 kind: VolumeSnapshotLocation
 metadata:
   name: {{ include "velero.volumeSnapshotLocation.name" . }}
+  annotations:
+    "helm.sh/hook": post-install
```
Comment (jenting) [inline charts/velero/templates/volumesnapshotlocation.yaml:7]: [1]

File: charts/velero/templates/crds/cleanup-crds.yaml [hunk 3]
```diff
@@ -1,39 +0,0 @@
-# This job is meant primarily for cleaning up on CI systems.
```
Comment (jenting) [inline charts/velero/templates/crds/cleanup-crds.yaml:1]: I think it would be good to keep this file since it supports delete CRDs.

Comment (carlisia) [reply charts/velero/templates/crds/cleanup-crds.yaml:1]: Yes, I agree. We don't need to keep it inside the `crds` folder I don't think.

For reference:
https://github.com/vmware-tanzu/helm-charts/blob/efbcab45316125c5760a63cc75f2343879cf3fc0/charts/velero/ci/test-values.yaml#L26

**Comment by carlisia** [comment]
I'm testing this PR today.

File: charts/velero/templates/crds/cleanup-crds.yaml (hunk 3 above)
Comment (tomahawk28) [reply charts/velero/templates/crds/cleanup-crds.yaml]: Good point, I revert it and placed in under `templates`

**Comment by tomahawk28** [comment]
@jenting @carlisia PTAL!

**Comment by sebastien-prudhomme** [comment]
@tomahawk28 for Helm2, this commit breaks tools such are Argo CD that are using some king of "helm template" without using Tiller service.

**Comment by tom-hadlaw-hs** [comment]
> @tomahawk28 for Helm2, this commit breaks tools such are Argo CD that are using some king of "helm template" without using Tiller service.

I'm having the same issue but for Helm3

**Pull request merged**